### ISSUE TITLE
isort ability to treat comments as code

### ISSUE THREAD

**Issue by yasirroni** [astral-sh/ruff#9697]
isort support ability to treat comments as code https://github.com/pycqa/isort/issues/1357. It is very useful in `.py` files generated from `.ipynb` that each section of import separated by comment `# %%`

Thus, 
```python
import os

import cvxpy as cp
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

# %%  >>> This should make all import below this treated as separate isort task
import sys
import traceback
import warnings
```

**Comment by yasirroni** [comment]
That would make sense. I just need to exclude all generated py file. Thanks.

**Comment by charliermarsh** [comment]
I think I'd prefer not to support this until we see a compelling reason to do so, since we support import sorting within notebooks natively. Gonna close for now.

**Comment by mwoitek** [comment]
I've found this issue because I'd like to have this feature. So I want to give you a reason to add it. I don't like to use Jupyter notebooks for coding. I avoid them with the help of a tool named Jupytext. It allows me to create notebooks from Python scripts. To do so, I add to my scripts special comments that serve as cell delimiters. Then it's important that, when I format my code, these comments remain in place. That's precisely the problem that isort's `treat_comments_as_code` solves. Please consider adding this option.
